File: src/Game/Configs/UnitConfig.lua
local UnitConfig = {}

---@class UnitConfig
---@field UTID int
---@field MainAtt int
---@field HP int
---@field MP int
---@field ATK1 int
---@field ATK2 int
---@field Def float
---@field SDef float
---@field Dodge float
---@field Crit float
---@field Career int

---@type UnitConfig[]
UnitConfig.DB = {
    { UTID = UTID_BLADE_MASTER , MainAtt = ATT_STR, HP = 3000  , MP = 200, ATK1 = 50  , ATK2 = 100  , Def = 0.1, SDef = 1 , Dodge = 0.25    ,  Crit = 0.1    , Career = CAREER_TYPE_DPS   },
}

local ref = {}
for _, v in pairs(UnitConfig.DB) do
    ref[v.UTID] = v
end
---@type table<int, UnitConfig>
UnitConfig.ById = setmetatable(ref, {
    __index = function(t, k)
        local v = rawget(t, k)
        if v then
            return v
        else
            logerror("id not found " .. k)
        end
    end
})

return UnitConfig

File: src/Game/Configs/UnitConfig.ts
export enum CareerType {
    Tank = 1,
    Healer = 2,
    DPS = 4,
    Minion = 8,
    Boss = 8,
    Creep = 32,
}

export enum AttributeType {
    None = 0,
    Strength = 1,
    Agility = 2,
    Intelligence = 3,
}

export interface IUnitConfig {
    UTID: number;
    PrimAtt: AttributeType;
    Str: number;
    Agi: number;
    Int: number;
    HP: number;
    MP: number;
    ATK1: number;
    ATK2: number;
    Def: number;
    SDef: number;
    Dodge: number;
    Crit: number;
    CritEx: number;
    Career: CareerType;
}

export const UnitConfig: { [key: number]: IUnitConfig } = {
    [1331850337]: { UTID: 1331850337, PrimAtt: AttributeType.Strength, Str: 15, Agi: 10, Int: 5, HP: 700, MP: 50, ATK1: 50, ATK2: 100, Def: 0.1, SDef: 0.05, Dodge: 0.25, Crit: 0.1, CritEx: 1.1, Career: CareerType.DPS },
    [1164207469]: { UTID: 1164207469, PrimAtt: AttributeType.Agility, Str: 9, Agi: 12, Int: 9, HP: 600, MP: 300, ATK1: 30, ATK2: 70, Def: 0.15, SDef: 0.1, Dodge: 0.35, Crit: 0.2, CritEx: 1, Career: CareerType.DPS },
    [1214931305]: { UTID: 1214931305, PrimAtt: AttributeType.Intelligence, Str: 7, Agi: 5, Int: 18, HP: 600, MP: 600, ATK1: 20, ATK2: 80, Def: 0.05, SDef: 0.2, Dodge: 0.03, Crit: 0.05, CritEx: 1, Career: CareerType.DPS },
};
